OpenLaTeXStudio/EditorModule/src/latexstudio/editor/EditorTopComponent.java

```diff
@@ -199,11 +199,40 @@ public DbxState getDbxState() {
     public void setDbxState(DbxState dbxState) {
         this.dbxState = dbxState;
     }
-    
+
     public String getLatexPath() {
         return latexPath;
     }
-    
+
+    public void commentOutText() {
+        String highlightedTextArea = rSyntaxTextArea.getSelectedText();
+        
+        try {
+            if (highlightedTextArea != null) {
+                if (highlightedTextArea.startsWith("%")) {
+                    rSyntaxTextArea.replaceSelection(highlightedTextArea.replace("%", ""));
+
+                } else {
+                    rSyntaxTextArea.replaceSelection(highlightedTextArea.replace("\n", "\n%"));
```

Comment:
This does not comment the first line of my selection. So if I select lines 4, 5, 6 and click toggle comment, lines 5 and 6 are commented out.